influunt-api/modules/influunt-core/app/models/Plano.java

```diff
@@ -259,6 +269,18 @@ public boolean isModoOperacaoVerde() {
         return Objects.nonNull(getModoOperacao()) && !ModoOperacaoPlano.APAGADO.equals(getModoOperacao()) && !ModoOperacaoPlano.INTERMITENTE.equals(getModoOperacao());
     }
 
+    @AssertTrue(groups = PlanosCheck.class, message = "Este plano deve ser configurado em todos os aneis.")
+    public boolean isPlanoPresenteEmTodosOsAneis() {
+        if (getAnel() != null && getAnel().isAtivo()) {
+            return this.getAnel().getControlador().getAneisAtivos().stream().allMatch(anel -> {
+                return anel.getPlanos().stream()
+                    .anyMatch(plano -> this.isTemporario() || (plano != null && Objects.equals(plano.posicao, this.posicao)));
```

Comment:
Não seria melhor colocar o `if (this.isTemporario())` no `if` mais externo, para evitar processamento?